File: sort/bubble_sort.py
def bubbleSort(nums: list):
    n = len(nums)
    for i in range(0, n - 1):
        for j in range(0, n - i - 1):
            if (nums[j] > nums[j + 1]):
                nums[j], nums[j + 1] = nums[j + 1], nums[j]
        print(nums)

def bubbleSort2(nums: list):
    n = len(nums)
    for i in range(0, n):
        for j in range(i + 1, n - i):
            if (nums[j] < nums[j - 1]):
                nums[j], nums[j - 1] = nums[j - 1], nums[j]
        print(nums)

if __name__ == "__main__":
    nums = [10, 3, 1, 8, 4,  7, 6]
    print("Before Sorted: ", nums)
    bubbleSort(nums)
    print("After Sorted: ", nums)
        

File: sort/bubble_sort.cpp
#include <vector>
#include <iostream>
using namespace std;

void printVector(const vector<int>& nums) {
    for (const int& num : nums) {
        cout << num << " ";
    }
    cout << endl;
}

// 本质上是选择排序
void bubbleSort(vector<int>& nums) {
    for (size_t i{0UL}; i < nums.size(); ++i) {
        for (size_t j = i + 1; j < nums.size(); ++j) {
            if (nums[i] > nums[j]) {
                swap(nums[j], nums[i]);
            }
        }
        printVector(nums);
    }
}

void BubbleSort(vector<int>& nums) {
    for (size_t i{0UL}; i < nums.size() - 1; ++i) {
        for (size_t j = 0; j < nums.size() - 1 - i; ++j) {
            if (nums[j + 1] < nums[j]) {
                swap(nums[j], nums[j + 1]);
            }
        }
        printVector(nums);
    }
     
}

int main(int argc, char const *argv[])
{
    std::vector<int> nums{2, 3, 1, 5, 4, 7, 6};
    cout << "Original nums: ";
    printVector(nums);
    BubbleSort(nums);
    cout << "After Sorted: ";
    printVector(nums);
    return 0;
}
